### PR TITLE
Improve the contribution guidelines for developers

### PR DESCRIPTION
**Pull request by deepyaman**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
Trivial improvements as I was reading the contribution guidelines

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/2171"></a>

### CODE REVIEW

File: docs/source/contribution/developer_contributor_guidelines.md [hunk 1]
```diff
@@ -116,19 +116,19 @@ We will work with you to complete your contribution, but we reserve the right to
 
 ## Create a pull request
 
-Create your pull request with [a descriptive title](#pull-request-title-conventions). Before you submit it, consider the following:
+Create your pull request with a [descriptive title](#pull-request-title-conventions). Before you submit it, consider the following:
```
Comment (stichbury) [inline docs/source/contribution/developer_contributor_guidelines.md:119]:
```suggestion
[Give your pull request a descriptive title](#pull-request-title-conventions). Before you submit it, consider the following:
```

Comment (stichbury) [reply docs/source/contribution/developer_contributor_guidelines.md:119]: Can I suggest this link as most descriptive/accessible?

**Review by stichbury — APPROVED** [review]
Nice fixes in here, thanks!

One minor comment but all good from my side. I expect Juan Luis will have a review/revision of this page when he joins us.

**Review by merelcht — APPROVED** [review]
👍

**Comment by deepyaman** [comment]
> Nice fixes in here, thanks!
> 
> One minor comment but all good from my side. I expect Juan Luis will have a review/revision of this page when he joins us.

@stichbury Accepted your suggestion, but not sure if I understood your comment:

> Can I suggest this link as most descriptive/accessible?

Was there some other action past your suggestion here?

File: docs/source/contribution/developer_contributor_guidelines.md [hunk 2]
```diff
@@ -116,19 +116,19 @@ We will work with you to complete your contribution, but we reserve the right to
 
 ## Create a pull request
 
-Create your pull request with [a descriptive title](#pull-request-title-conventions). Before you submit it, consider the following:
+[Give your pull request a descriptive title](#pull-request-title-conventions). Before you submit it, consider the following:
 
 * You should aim for cross-platform compatibility on Windows, macOS and Linux
 * We use [SemVer](https://semver.org/) for versioning
```
Comment (deepyaman) [inline docs/source/contribution/developer_contributor_guidelines.md:122]:
```suggestion
* We use [Semantic Versioning](https://semver.org/) for versioning
```
When I was borrowing from these contribution guidelines for internal use, I got a recommendation to spell out Semantic Versioning; I think it makes sense?

**Comment by stichbury** [comment]
> Was there some other action past your suggestion here?

Sorry, no, just the comment that including more words in the link more extensively made it more accessible. If you accepted the suggestion, that's all that's need.

**Review by stichbury — APPROVED** [review]
@deepyaman Do you want me to merge this (or will you?). I think it's good to go 🏆

**Comment by deepyaman** [comment]
> @deepyaman Do you want me to merge this (or will you?). I think it's good to go 🏆

I'll do it as soon as build passes, thanks! (And if I forget, feel free.)

**Pull request merged**